### PR TITLE
JBIDE-27060 ODO OpenShift ITests

### PR DESCRIPTION
**Pull request by jkopriva**
Signed-off-by: Josef Kopriva <jkopriva@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by jkopriva** [comment]
@odockal No Jenkins job yet. I will create new PR with stabilisation fixes for Jenkins and new tests(I didn't want to have big PR).

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/OpenShiftApplicationExplorerView.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			return false;
+		}
+	}
+
+	public OpenShiftODOConnection getOpenShiftODOConnection() {
+		OpenShiftODOConnection connection = null;
+
+		connection = new OpenShiftODOConnection(getConnectionItem());
+
+		assertNotNull("Unable to get OpenShift connection.", connection);
```
Comment (odockal) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/OpenShiftApplicationExplorerView.java:146]: Why to use assert in test framework? Wouldn't it cleaner to use throwing an exception and deal with it on the usage, ie. line: 121?

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	public OpenShiftODOProject createNewProject(String projectName) {
+		select();
+		new ContextMenuItem(OpenShiftLabel.ContextMenu.NEW_PROJECT).select();
+		
+		InputDialog dialog = new InputDialog(OpenShiftLabel.Shell.NEW_PROJECT);
+		dialog.setInputText(projectName);
+		dialog.ok();
+		
+		new WaitWhile(new ShellIsAvailable(OpenShiftLabel.Shell.NEW_PROJECT), TimePeriod.LONG);
+		new WaitWhile(new JobIsRunning(), TimePeriod.LONG);
```
Comment (odockal) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java:98]: I see a lot of JobIsRunning usage, but if you got some special job for this purpose I would say it is much better to implement specific job wait condition or pass a matcher into JobIsRunning instantiaon in order to wait for specific job.

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	 *  
+	 * @param projectName name of a project, if displayed name is provided,
+	 *  	its displayed name, otherwise its project name
+	 * @return OpenShift project
+	 */
+	public OpenShiftODOProject getProject(String projectName) {
+		activateOpenShiftApplicationExplorerView();
+		item.expand();
+		
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
```
Comment (odockal) [inline test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java:52]: VERY_LONG timeout for item expand? Isn't that quite a bit? Also, if there is specific job, use its name to wait for.

File: itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/AbstractODOTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * @author jkopriva@redhat.com
+ */
+public abstract class AbstractODOTest {
+	
+	private static String eclipseProject = "myService";
+	
+	@After
+	public void cleanUp() {
+		//Cleanup notifications
+		WorkbenchShellHandler.getInstance().closeAllNonWorbenchShells();
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/AbstractODOTest.java:37]: There are some notifications after creating project, so the test fails with error(some window is opened - its notification).
So the test fails, even if the project is successfully created.

File: itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/odo/ConnectionODOCommandsTests.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		new WaitUntil(new ConsoleHasText("Services available through Service Catalog"));
+	}
+	
+	@Test
+	public void testListCatalogComponents () {
+		connection.listCatalogComponents();
+		new WaitUntil(new ConsoleHasText("Odo OpenShift Components:"));
+	}
+	
+	//@Test
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/odo/ConnectionODOCommandsTests.java:54]: +1

File: itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/AbstractODOTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (odockal) [reply itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/AbstractODOTest.java:37]: alright then.

File: itests/org.jboss.tools.openshift.ui.bot.test/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -23,7 +23,8 @@ Require-Bundle: org.eclipse.ui.ide,
  org.jboss.tools.openshift.core,
  org.jboss.tools.openshift.ui,
  org.eclipse.linuxtools.docker.reddeer,
- org.eclipse.reddeer.common
+ org.eclipse.reddeer.common,
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply itests/org.jboss.tools.openshift.ui.bot.test/META-INF/MANIFEST.MF]: removed

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftODOConnectionRequirement.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+					explorer.connectToOpenShiftODOBasic(server, username, password);
+				} else if (StringUtils.isNotEmpty(token)) {
+					explorer.connectToOpenShiftODOOAuth(server, token);
+				} else {
+					throw new OpenShiftToolsException("Could not create connection");
+				}
+			}
+
+		} catch (Exception s) {
+			throw new OpenShiftToolsException("Cannot fullfill requirement");
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftODOConnectionRequirement.java]: fixed

File: test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java (hunk 3 above)
Comment (jkopriva) [reply test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java]: changed to LONG

**Comment by jkopriva** [comment]
@odockal can you please review this once again? All issues should be now fixed.

**Pull request merged**